sentence_transformers/sparse_encoder/models/CSRSparsity.py

```diff
@@ -0,0 +1,210 @@
+        self.k_aux = k_aux
+        self.normalize = normalize
+
+        self.stats_last_nonzero: torch.Tensor
+        self.register_buffer("stats_last_nonzero", torch.zeros(hidden_dim, dtype=torch.long))
+
+        def auxk_mask_fn(x):
+            dead_mask = self.stats_last_nonzero > dead_threshold
+            x.data *= dead_mask  # inplace to save memoryr
+            return x
+
+        self.auxk_mask_fn = auxk_mask_fn
+
+    def encode_pre_act(self, x: torch.Tensor) -> torch.Tensor:
+        """
+        :param x: input data (shape: [batch, input_dim])
+        :param latent_slice: slice of latents to compute
+            Example: latent_slice = slice(0, 10) to compute only the first 10 latents.
+        :return: autoencoder latents before activation (shape: [batch, hidden_dim])
+        """
+        x = x - self.pre_bias
+        latents_pre_act = F.linear(x, self.encoder.weight, self.latent_bias)
+        return latents_pre_act
+
+    def LN(self, x: torch.Tensor, eps: float = 1e-5):
+        mu = x.mean(dim=-1, keepdim=True)
+        x = x - mu
+        std = x.std(dim=-1, keepdim=True)
+        x = x / (std + eps)
+        return x, mu, std
+
+    def preprocess(self, x: torch.Tensor):
+        if not self.normalize:
+            return x, dict()
+        x, mu, std = self.LN(x)
+        return x, dict(mu=mu, std=std)
+
+    def top_k(self, x: torch.Tensor, k=None) -> torch.Tensor:
+        """
+        :param x: input data (shape: [batch, input_dim])
+        :return: autoencoder latents (shape: [batch, hidden_dim])
+        """
+        if k is None:
+            k = self.k
+        topk = torch.topk(x, k=k, dim=-1)
+        z_topk = torch.zeros_like(x)
+        z_topk.scatter_(-1, topk.indices, topk.values)
+        latents_k = F.relu(z_topk)
+        ## set num nonzero stat ##
+        tmp = torch.zeros_like(self.stats_last_nonzero)
+        tmp.scatter_add_(
+            0,
+            topk.indices.reshape(-1),
+            (topk.values > 1e-5).to(tmp.dtype).reshape(-1),
+        )
+        self.stats_last_nonzero *= 1 - tmp.clamp(max=1)
+        self.stats_last_nonzero += 1
+        ## end stats ##
+
+        if self.k_aux:
+            aux_topk = torch.topk(
+                input=self.auxk_mask_fn(x),
+                k=self.k_aux,
+            )
+            z_auxk = torch.zeros_like(x)
+            z_auxk.scatter_(-1, aux_topk.indices, aux_topk.values)
+            latents_auxk = F.relu(z_auxk)
+        return latents_k, latents_auxk
+
+    def decode(self, latents: torch.Tensor, info=None) -> torch.Tensor:
+        """
+        :param latents: autoencoder latents (shape: [batch, hidden_dim])
+        :return: reconstructed data (shape: [batch, n_inputs])
+        """
+
+        ret = self.decoder(latents) + self.pre_bias
+
+        if self.normalize:
+            assert info is not None
+            ret = ret * info["std"] + info["mu"]
+        return ret
+
+    def forward(self, features: torch.Tensor) -> torch.Tensor:
```

Comment:
```suggestion
    def forward(self, features: torch.Tensor, k: int | None = None) -> torch.Tensor:
        k = k if k is not None else self.k
```
For example, then `encode` can pass down parameters via the `module_kwargs`.